crates/router/src/core/payments/transformers.rs

```diff
@@ -2431,3 +2499,220 @@ impl ForeignFrom<CustomerDetails> for router_request_types::CustomerDetails {
         }
     }
 }
+
+#[cfg(feature = "v2")]
+impl ForeignFrom<api_models::payments::AmountDetails>
+    for hyperswitch_domain_models::payments::AmountDetails
+{
+    fn foreign_from(amount_details: api_models::payments::AmountDetails) -> Self {
+        Self {
+            order_amount: amount_details.order_amount.into(),
+            currency: amount_details.currency,
+            shipping_cost: amount_details.shipping_cost,
+            tax_details: Some(diesel_models::TaxDetails {
+                default: amount_details
+                    .order_tax_amount
+                    .map(|order_tax_amount| diesel_models::DefaultTax { order_tax_amount }),
+                payment_method_type: None,
+            }),
+            skip_external_tax_calculation:
+                hyperswitch_domain_models::payments::TaxCalculationOverride::foreign_from(
+                    amount_details.skip_external_tax_calculation,
+                ),
+            skip_surcharge_calculation:
+                hyperswitch_domain_models::payments::SurchargeCalculationOverride::foreign_from(
+                    amount_details.skip_surcharge_calculation,
+                ),
+            surcharge_amount: amount_details.surcharge_amount,
+            tax_on_surcharge: amount_details.tax_on_surcharge,
+        }
+    }
+}
+
+#[cfg(feature = "v2")]
+impl ForeignFrom<hyperswitch_domain_models::payments::AmountDetails>
+    for api_models::payments::AmountDetails
+{
+    fn foreign_from(amount_details: hyperswitch_domain_models::payments::AmountDetails) -> Self {
+        Self {
+            order_amount: amount_details.order_amount.into(),
+            currency: amount_details.currency,
+            shipping_cost: amount_details.shipping_cost,
+            order_tax_amount: amount_details.tax_details.and_then(|tax_details| {
+                tax_details.default.map(|default| default.order_tax_amount)
```

Comment:
Avoid using key words, or function name in std library
```suggestion
                optional_tax_details.default.map(|tax_details| tax_details.order_tax_amount)
```